### PR TITLE
Installation instructions for MacOS and Cloud

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
1. Added installation instructions for - MacOS + CPU + Python 
2. Added links to AWS AMI and CFN. MXNet on Cloud.
3. Tested all the instructions.

Note: I am not fully happy with the size/structure of this file getting bigger and it becomes more complex to manage when we add other Programming language installation guide. I will create an issue and assign it to myself to simplify this post v0.10 doc release.

@madjam @zackchase @lxn2 - Can you please review this changes?

### CODE REVIEW

File: docs/get_started/install.md [hunk 1]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+
+Install *MXNet* with OpenBLAS acceleration.
+
+```bash
+(mxnet)$ pip install mxnet
+```
+
+**Step 5**  Validate the installation by running simple *MXNet* code described [here](#validate-mxnet-installation).
+
+**Note**  You can read more about virtualenv [here](https://virtualenv.pypa.io/en/stable/userguide/).
```
Comment (mli) [inline docs/get_started/install.md]: should we use virtualenv. i don't think it is necessary given our light dependencies. we can just let user run `pip install mxnet`. that's, remove the virtualenv section to keep doc short

File: docs/get_started/install.md [hunk 2]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+</div>
+
+<div class="docker">
+<br/>
+
+Docker images with *MXNet* are available at [Docker Hub](https://hub.docker.com/r/mxnet/).
+
+**Step 1**  Install Docker on your machine by following the [docker installation instructions](https://docs.docker.com/docker-for-mac/install/#install-and-run-docker-for-mac).
+
+*Note* - You can install Community Edition (CE) to get started with *MXNet*.
```
Comment (mli) [inline docs/get_started/install.md]: what does it mean?

File: docs/get_started/install.md [hunk 3]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+
+```bash
+$ curl -O https://raw.githubusercontent.com/dmlc/mxnet/master/setup-utils/install-mxnet-osx-python.sh
+```
+
+**Step 2** Run the script to get latest MXNet source and build.
+
+```bash
+# Make the script executable
+$ sudo chmod 744 install-mxnet-osx-python.sh
```
Comment (mli) [inline docs/get_started/install.md]: just `chmod +x xxx.sh` no sudo is required. suggest to make these two as a single line to make users easier to copy

chmod +x xx.sh && ./xx.sh

File: docs/get_started/install.md [hunk 4]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+
+# Run the script. It takes around 5 mins.
+$ bash install-mxnet-osx-python.sh
+```
+
+**Step 3** Validate the installation by running simple MXNet code described [here](#validate-mxnet-installation).
+
+</div>
+
+</div>
```
Comment (mli) [inline docs/get_started/install.md]: change to `</div> <!-- end of pip -->`

File: docs/get_started/install.md [hunk 5]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+</div>
+</div>
+
+<!-- END - Mac OS Python CPU Installation Instructions -->
```
Comment (mli) [inline docs/get_started/install.md]: not necessary if resolved the above comment

File: docs/get_started/install.md [hunk 6]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+
+<!-- START - Mac OS Python GPU Installation Instructions -->
+
+<div class="macos">
+  <div class="python">
+    <div class="gpu">
```
Comment (mli) [inline docs/get_started/install.md]: change to `<div class="macos python gpu">.` split it later if more staffs are added

File: docs/get_started/install.md [hunk 7]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+(mxnet)$ pip install --upgrade setuptools
+```
+
+Install *MXNet* with OpenBLAS acceleration.
+
+```bash
+(mxnet)$ pip install mxnet
+```
+
+**Step 5**  Validate the installation by running simple *MXNet* code described [here](#validate-mxnet-installation).
```
Comment (lxn2) [inline docs/get_started/install.md]: What do you think about installing docker with homebrew? That way we have one more requirement that we can give concrete commands for.

File: docs/get_started/install.md [hunk 8]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+
+**Step 1**  Install prerequisites - Homebrew, python development tools.
+
+```bash
+# Install Homebrew
+$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+$ export PATH=/usr/local/bin:/usr/local/sbin:$PATH
+
+# Install python development tools - python2.7, pip, python-setuptools
+$ brew install python
```
Comment (lxn2) [inline docs/get_started/install.md]: Does this one command actually install all of python, pip, and setuptools?

File: docs/get_started/install.md [hunk 9]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+**Step 3**  Validate the installation by running simple MXNet code described [here](#validate-mxnet-installation).
+
+</div>
+
+<div class="docker">
+<br/>
+
+Docker images with *MXNet* are available at [Docker Hub](https://hub.docker.com/r/mxnet/).
+
+**Step 1**  Install Docker on your machine by following the [docker installation instructions](https://docs.docker.com/docker-for-mac/install/#install-and-run-docker-for-mac).
```
Comment (lxn2) [inline docs/get_started/install.md]: What do you think about installing docker with homebrew? That way we have one more requirement that we can give concrete commands for.

**Comment by lxn2** [comment]
Looks good to me. I like the way it's currently organized with docker/virtualenv/build from source.  Though possibly the virtualenv & pip instructions are almost identical and I wonder if both are necessary.

File: docs/get_started/install.md (hunk 1 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: I agree for now, we have only mxnet, but, I see other dependencies like jupyter, graphviz etc.. will be required. My idea is to evolve this to have a requirement file, which user can use in virtualenv to get complex mxnet environment setup in one command.

Also, for user, we are showing options in buttons, so should not complicate with length.

File: docs/get_started/install.md (hunk 2 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: There are 2 options in docker installation page - Community Edition & Enterprise Edition, I wanted to state - Community Edition is good to start with MXNet. Maybe I should reword it?

File: docs/get_started/install.md (hunk 6 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Right. Wanted to keep it more uniform across the page, so it is easier for new devs to modify and maintain.

File: docs/get_started/install.md (hunk 8 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Yes.

File: docs/get_started/install.md (hunk 9 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Makes sense but I think with homebrew we get DockerToolbox. I felt, Docker installation page for mac is pretty simple - download and install dmg file. hence redirected users there.

File: docs/get_started/install.md [hunk 10]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+<!-- START - MacOS Python CPU Installation Instructions -->
 
-# Validate MXNet Installation
+<div class="macos">
+  <div class="python">
+    <div class="cpu">
+
+The following installation instructions have been tested on OSX Sierra and El Capitan.
+
+
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Fixed.

File: docs/get_started/install.md [hunk 11]
```diff
@@ -424,13 +443,214 @@ $ sudo python setup.py install
...
+<div class="virtualenv">
+<br/>
+
+**Step 1**  Install prerequisites - Homebrew, python development tools.
+
+```bash
+# Install Homebrew
+$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+$ export PATH=/usr/local/bin:/usr/local/sbin:$PATH
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Yes. This is a transient issue with PATH + homebrew installed packages. Setting these paths at the start of PATH, will help to make it more deterministic and fool proof.

File: docs/get_started/install.md (hunk 3 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Removed sudo.

I agree it reduces number of times users need to copy paste. But, I felt this is more clear for user on what is being done.

File: docs/get_started/install.md (hunk 4 above)
Comment (sandeep-krishnamurthy) [inline docs/get_started/install.md]: Done.

**Comment by sandeep-krishnamurthy** [comment]
Fixed Comments by Mu.

@mli - I agree Re-arranging helps to reduce the content size, but, for automated testing and maintaining this page would be harder.
For users, it looks simpler. But, as I mentioned earlier and discussed with you offline - I am not fully happy with this big file. I will create an issue and assign it to myself. To better organize content on this page. I will take it up post v0.10 release. As this is solving the customers problem. I think we are good to go for this release and optimize it later.

**Comment by lxn2** [comment]
@mli if this looks good can you merge it? I am almost done with the test script for this and hope to add a Jenkins job by end of today or tomorrow.

**Comment by sandeep-krishnamurthy** [comment]
@zackchase @madjam - Can you please this PR?

**Comment by piiswrong** [comment]
This is too long. Its as simple as a `pip install mxnet` for most users and that's the whole point. We shouldn't make it appear more complicated than it is.

We should keep the original one sentence instruction. You can put this in a trouble shooting page if necessary.

**Comment by piiswrong** [comment]
I'll merge this first since I don't see it being linked to from any page. but we need to clean it up later.

**Comment by sandeep-krishnamurthy** [comment]
@piiswrong  - I agree source is too long and I had same thoughts in my PR description. However, when it renders, I felt it is very easy and complete for readers to install required pre-requistes, mxnet and at the end test with a small code snippet.

Can you please go through the instructions here - http://mxnet.io/get_started/install.html and let me know if you think it is complicated for readers to install and get started. I can take it up for  simplification.

We can have in home page just one line - pip install mxnet and one line docker image command.

**Pull request merged**